contracts/light_client/tests/client_test.rs

```diff
@@ -189,23 +189,375 @@ mod light_client {
             let block308 = file_as_json::<Block>("308.json").unwrap();
 
-            let context_244 =
-                get_context(accounts(0), 244 * TEST_BLOCK_TIMESTAMP_MULTIPLIER, 244);
+            let context_244 = get_context(accounts(0), 244 * TEST_BLOCK_TIMESTAMP_MULTIPLIER, 244);
             testing_env!(context_244.build());
             bridge.init_with_validators(initial_validators);
 
-            let context_304 =
-                get_context(accounts(0), 304 * TEST_BLOCK_TIMESTAMP_MULTIPLIER, 304);
+            let context_304 = get_context(accounts(0), 304 * TEST_BLOCK_TIMESTAMP_MULTIPLIER, 304);
             testing_env!(context_304.build());
             bridge.init_with_block(block304);
 
             bridge.set_paused(PAUSE_ADD_BLOCK_HEADER);
 
             // switch context to accounts(1) which is not the admin account, and let that account try to add a block
-            let context_308 =
-                get_context(accounts(1), 308 * TEST_BLOCK_TIMESTAMP_MULTIPLIER, 308);
+            let context_308 = get_context(accounts(1), 308 * TEST_BLOCK_TIMESTAMP_MULTIPLIER, 308);
             testing_env!(context_308.build());
             bridge.add_light_client_block(block308);
         }
+
+        #[test]
+        fn test_rel46() {
+            let mut bridge = init(None);
+            let block105190359 = file_as_json::<Block>("block_105190359.json").unwrap();
+            let initial_validators = block105190359.next_bps.unwrap();
+            let context_105190359 = get_context(
+                accounts(0),
+                105190359 * TEST_BLOCK_TIMESTAMP_MULTIPLIER,
+                105190359,
+            );
+            testing_env!(context_105190359.build());
+            bridge.init_with_validators(initial_validators);
+            let block105233559 = file_as_json::<Block>("block_105233559.json").unwrap();
+            let context_105233559 = get_context(
+                accounts(0),
+                105233559 * TEST_BLOCK_TIMESTAMP_MULTIPLIER,
+                105233559,
+            );
+            testing_env!(context_105233559.build());
+            bridge.init_with_block(block105233559);
+            assert!(
+                encode_hex(&bridge.block_merkle_roots(105233559).unwrap())
+                    == "c348cdd4dfa14b1fcdff6688dd2321b2451c237f9eb38ee3353a724b832bb3f6"
+            );
+            let block105276759 = file_as_json::<Block>("block_105276759.json").unwrap();
+            let context_105276759 = get_context(
+                accounts(0),
+                105276759 * TEST_BLOCK_TIMESTAMP_MULTIPLIER,
+                105276759,
+            );
+            testing_env!(context_105276759.build());
+            bridge.add_light_client_block(block105276759);
+            assert!(
+                encode_hex(&bridge.block_merkle_roots(105276759).unwrap())
+                    == "f5bd2431608fa4b190511d90d38032e129d8525725207e452eecebf737db3b1f"
+            );
```

Comment:
consider doing add_light_client_block in a loop, the only thing that changes in these calls is under the assertion, the expected block_merkle_root which could be extracted in an array or map of expected_block_merkle_roots.
Don't feel strong about it, leaving it to you

Reply:
also, add a short comment what is achieved with this test, e.g. 'adding blocks from at least 2 different epochs..'